Python/Product/Analysis/ExpressionFinder.cs

```diff
@@ -162,37 +188,29 @@ private class NormalExpressionWalker : ExpressionWalker {
                     Expression = node;
                 }
 
-                node.NameExpression?.Walk(this);
+                if (_options.ClassDefinitionName) {
+                    node.NameExpression?.Walk(this);
+                }
                 node.Decorators?.Walk(this);
-                _walkingClassBases = true;
-                try {
-                    foreach (var b in node.Bases.MaybeEnumerate()) {
-                        b.Walk(this);
-                    }
-                } finally {
-                    _walkingClassBases = false;
+                foreach (var b in node.Bases.MaybeEnumerate()) {
+                    b.Walk(this);
                 }
                 node.Body?.Walk(this);
 
-                return true;
+                return false;
             }
 
             public override bool Walk(Arg node) {
```

Comment:
So this one always return `false`?

Reply:
`false` here means don't continue walking down this branch. In this case, it's because we've already walked it (the `node.Body.Walk(this)` above).